src/core/include/units/bits/prime.h

```diff
@@ -134,7 +134,7 @@ struct wheel_factorizer {
   static constexpr auto coprimes_in_first_wheel =
     coprimes_up_to<num_coprimes_up_to(wheel_size, basis)>(wheel_size, basis);
 
-  static constexpr std::size_t find_first_factor(std::size_t n)
+  static constexpr std::intmax_t find_first_factor(std::intmax_t n)
```

Comment:
@chiphogg, could you please confirm that (i.e. is it OK or no other similar changes are needed in other functions to be consistent)?